backend/src/project/endpoints.ts

```diff
@@ -49,11 +78,15 @@ export const saveProject = async (req: Request, res: Response) => {
         res.status(400).json({ error: "Invalid request body: " + data });
         return;
     }
+    if (!req.user) {
```

Comment:
Ja bym nie dodawał tej walidacji wewnatrz tej funkcji. Jak chcemy bezwarunkowo zablokować jakiś endpoint dla niezalogowanych użytkowników, można przygotować taką funkcję isLoggedInd i ją w prosty sposób dodawać do endpointów, które chcemy zablokować dla niezalogowanego użytkownika.

Reply:
I to generalnie przydałoby się obsłużyć na frontendzie, żeby poinformować użytkownika, że żeby zapisać projekt to musi sie zalogować. Jak wolisz działać w backendzie niż na frontendzie to daj znać, to to można przydzielić komuś innemu

Reply:
Wolę backend jeśli mam być szczery, więc można to wydzielić jako osobny task. Nigdy we frontendach nie siedziałem zbytnio, więc żeby to zrobić to bym potrzebował ogarnąć co jak działa, a nie chcę przeciągać ukończenia tego jeszcze bardziej